constellations/gemini-admin/src/utils/inputText.js

```diff
@@ -0,0 +1,76 @@
+"use strict";
+require("./inputText.sass");
+import React from "react";
+import Checkbox from "../../assets/checkmark_green.svg";
+
+export class InputText extends React.Component {
+    render = () => {
+        const description = this.props.description;
+        const required = this.props.required;
+        const value = this.props.value;
+
+        var pass = true;
+        var error = <h4 className="hidden"></h4>;
+        if (required) {
+            this.props.validators.some((validator) => {
+                if (!validator.validate(value)) {
+                    error = <h4 className="red">{validator.message}</h4>;
+                    pass = false;
+                    return true;
+                }
+            });
+        } else {
+            pass = false;
+        }
+
+        var format = <h4 className="hidden"></h4>;
+        if (!!description) {
+            let ending;
+            if (required) {
+                if (pass) {
+                    ending = <span>{" (required)"}</span>;
+                } else {
+                    ending = <span className="red">{" (required)"}</span>;
+                }
+            } else {
+                ending = " (optional)";
+            }
```

Comment:
what do you think of a render_____() function to modularize this logic away from the main render() function?